src/algebra/big_operators/finprod.lean

```diff
@@ -736,6 +751,20 @@ over `a ∈ ⋃₀ t` is the product over `s ∈ t` of the products of `f a` ove
   ∏ᶠ a ∈ ⋃₀ t, f a = ∏ᶠ s ∈ t, ∏ᶠ a ∈ s, f a :=
 by rw [set.sUnion_eq_bUnion, finprod_mem_bUnion h ht₀ ht₁]
 
+lemma finprod_mem_dvd' (a : α) (hf : finite (mul_support f)) :
+  f a * (∏ᶠ i ∈ (mul_support f) \ {a}, f i) = ∏ᶠ i, f i  :=
```

Comment:
This name doesn't look right to me. `finset.mul_prod_erase` looks morally pretty similar (and probably leads to a shorter proof).

So maybe this should be `mul_finprod_sdiff_singleton`? And you should add `to_additive` to generate the `add_finsum` version.

Reply:
I changed the name to `finprod_mul_sdiff_singleton`. I tried rewriting the proof in terms of `finset.mul_prod_erase` and `finset.prod_erase` and it got a bit longer (but you may know how to golf it):

```
lemma finprod_mul_sdiff_singleton (a : α) (hf : finite (mul_support f)) :
  f a * (∏ᶠ i ∈ (mul_support f) \ {a}, f i) = ∏ᶠ i, f i  :=
begin
  classical,
  rw [finprod_eq_prod _ hf, finprod_mem_eq_finite_to_finset_prod _
    (finite.subset hf (diff_subset (mul_support f) {a}))],
  have : (finite.subset hf (diff_subset (mul_support f) {a})).to_finset = hf.to_finset \ {a},
  { ext,
    simp only [mem_singleton_iff, iff_self, finite.mem_to_finset, finset.mem_sdiff, mem_diff, 
    finset.mem_singleton],},
  rw [this, finset.sdiff_singleton_eq_erase],
  by_cases ha : a ∈ mul_support f,
  { apply finset.mul_prod_erase _ _ ((finite.mem_to_finset _ ).mpr ha), },
  { rw [mem_mul_support, not_not] at ha,
    rw [ha, one_mul],
    apply finset.prod_erase _ ha, }
end
```
So for now I have left the old proof.